samtranslator/model/sam_resources.py

```diff
@@ -2286,7 +2286,7 @@ def _construct_ddb_datasources(
         resources: List[Resource] = []
 
         for relative_id, ddb_datasource in ddb_datasources.items():
-            datasource_logical_id = self.logical_id + relative_id
+            datasource_logical_id = f"{self.logical_id}DynamoDBDataSources{relative_id}"
```

Comment:
nit, it refers to one specific source: 
```suggestion
            datasource_logical_id = f"{self.logical_id}DynamoDBDataSource{relative_id}"
```